geode-junit/src/main/java/org/apache/geode/internal/AvailablePortHelper.java

```diff
@@ -210,4 +188,100 @@ public static int getRandomAvailableUDPPort() {
     return getRandomAvailableUDPPorts(1)[0];
   }
 
+  public static void initializeUniquePortRange(int rangeNumber) {
```

Comment:
I think I'd prefer to have a fixed limit of, say, 100 ports per site, rather than this logic, which is more complicated and, if it ever breaks, will do so in confusing and unpredictable ways.

Reply:
I agree that this is significantly more complicated, but the benefit is that it gives each host the largest possible range of ports possible, without actually needing to know how many hosts there are or how many ports each one needs.  While it seems generally safe to assume that no test will ever need more than 100 ports, or 20 sites, it isn't necessary to introduce these limits, and it does put us more at risk of failing if we happen to running on machine where some other process took a large chunk of ephemeral address that happens to overlap with our range.

If we do end up in an environment where we do have to spill over into another host's range in order to fill a request we will have an nasty situation where we'll be racing with them in port assignments.  However, in order to end up in such a situation, we either have a crazy number of hosts, or have assigned a crazy number of ports from a particular host (likely both).  In this case, we'd likely have failed far sooner had we tried to impose limits on the number of hosts or ports.